spec/predicates/vsa.md

```diff
@@ -1,11 +1,279 @@
+>
+> Consumers MUST accept only specific (signer, verifier) pairs. For example,
+> "GitHub" can sign provenance for the "GitHub Actions" verifier, and "Google"
+> can sign provenance for the "Google Cloud Deploy" verifier, but "GitHub" cannot
+> sign for the "Google Cloud Deploy" verifier.
+>
+> The field is required, even if it is implicit from the signer, to aid readability and
+> debugging. It is an object to allow additional fields in the future, in case one
+> URI is not sufficient.
+
+<a id="verifier.id"></a>
+`verifier.id` _string ([TypeURI]), required_
+
+> URI indicating the verifier’s identity.
+
+<a id="verifier.version"></a>
+`verifier.version` _map (string->string), optional_
+
+> Map of names of components of the verification platform to their version.
+
+<a id="timeVerified"></a>
+`timeVerified` _string ([Timestamp]), optional_
+
+> Timestamp indicating what time the verification occurred.
+
+<a id="resourceUri"></a>
+`resourceUri` _string ([ResourceURI]), required_
+
+> URI that identifies the resource associated with the artifact being verified.
+
+<a id="policy"></a>
+`policy` _object ([ResourceDescriptor]), required_
+
+> Describes the policy that the `subject` was verified against.
+>
+> The entry MUST contain a `uri`.
+
+<a id="inputAttestations"></a>
+`inputAttestations` _array ([ResourceDescriptor]), optional_
+
+> The collection of attestations that were used to perform verification.
+> Conceptually similar to the `resolvedDependencies` field in [SLSA Provenance].
+>
+> This field MAY be absent if the verifier does not support this feature.
+> If non-empty, this field MUST contain information on _all_ the attestations
+> used to perform verification.
+>
+> Each entry MUST contain a `digest` of the attestation and SHOULD contains a
+> `uri` that can be used to fetch the attestation.
+
+<a id="verificationResult"></a>
+`verificationResult` _string, required_
+
+> Either “PASSED” or “FAILED” to indicate if the artifact passed or failed the policy verification.
+
+<a id="verifiedProperties"></a>
+`verifiedProperties` _array (_string), required_
+
+> Indicates the properties verified for the artifact (and not
+> its dependencies), or "FAILED" if policy verification failed.
+>
+> If including SLSA levels, users MUST NOT include more than one level per SLSA
+> track. Note that each SLSA level implies all levels below it (e.g.
+> SLSA_BUILD_LEVEL_3 implies SLSA_BUILD_LEVEL_2 and SLSA_BUILD_LEVEL_1), so
+> there is no need to include more than one level per track.
+
+<a id="dependencyProperties"></a>
+`dependencyProperties` _object, optional_
+
+> A count of the dependencies verified to have each property.
+>
+> Map from _string to the number of the artifact's _transitive_ dependencies
+> that were verified at the indicated level. Absence of a given value
+>  MUST be interpreted as reporting _0_ dependencies with that value.
+>
+> If including SLSA levels, users MUST count each dependency only once per SLSA
```

Comment:
I wonder if this text can be generalized.

> If including 'leveled' properties (e.g. SLSA), users MUST count each dependency once per property, at the highest level verified.

Reply:
That could work, but I prefer to delete this text (even if we keep the field) and incorporate it into the SLSA spec's description of how to use a VSA for SLSA.